### PR TITLE
Update build script to support Visual Studio 2022

### PR DESCRIPTION
**Pull request by emerick**
Resolves https://github.com/brave/brave-browser/issues/21636

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Comment by mihaiplesa** [comment]
Do we need any changes in https://github.com/brave/omaha/blob/main/omaha/hammer.bat?

**Comment by mherrmann** [comment]
@mihaiplesa yes, I'd think we at least need to change https://github.com/brave/omaha/blob/a4d777f2d4a6f77d50b80ff5aa3ff9d912a078cb/omaha/hammer.bat#L8.

**Review by mherrmann — COMMENTED** [review]
If we want to force VS 2022 throughout, then we would also need to update https://github.com/brave/omaha/blob/a4d777f2d4a6f77d50b80ff5aa3ff9d912a078cb/omaha/hammer.bat#L8.

**Review by mkarolin — APPROVED** [review]
++

**Review by mihaiplesa — APPROVED** [review]

File: build/commands/lib/util.js [hunk 1]
```diff
@@ -517,15 +517,20 @@ const util = {
     let msBuild = ''
     if (vsVersion === '2017') {
       msBuild = path.join(vsPath, 'MSBuild', '15.0', 'Bin', 'MSBuild.exe')
-    } else if (vsVersion === '2019') {
+    } else if (vsVersion === '2019' || vsVersion === '2022') {
```
Comment (bridiver) [inline build/commands/lib/util.js:520]: should this just be `} else {` because all versions > 2017 use Current?

Comment (emerick) [reply build/commands/lib/util.js]: Fixed up as discussed via Slack.

**Review by mihaiplesa — APPROVED** [review]

**Review by bridiver — APPROVED** [review]

**Pull request merged**